packages/@aws-cdk/aws-lambda-event-sources/lib/sqs.ts

```diff
@@ -29,14 +39,20 @@ export class SqsEventSource implements lambda.IEventSource {
   private _eventSourceMappingId?: string = undefined;
 
   constructor(readonly queue: sqs.IQueue, private readonly props: SqsEventSourceProps = { }) {
```

Comment:
Ideally you should also throw here `if (this.queue.fifo && this.props.maxBatchingWindow)`

Reply:
thanks for the suggestion @jogold, nice catch

adding [doc reference](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html) for posterity. 

> Batch window – Specify the maximum amount of time to gather records before invoking the function, in seconds. Only applicable to standard queues.